### ISSUE TITLE
[UI] Intro is always triggered on the latest element of a list

### ISSUE THREAD

**Issue by bub-bl** [Facepunch/sbox-issues#6768]
### Describe the bug

When you add a element in list within a panel, the intro is always triggered on the latest element of the list even you are adding an element at any index in of the list

### To Reproduce

- Create a component like that:
```html
@inherits PanelComponent

<root>
    @foreach(var item in _items)
    {
        <span class="any-style-with-an-intro-transition">@item</span>
    }
</root>

@code {

    private readonly List<object> _items = new();

    [Button("Add Item")]
    private void Add(object item)
    {
        _items.InsertAt(0, item);
    }

}
```
- Add items when the game is running, the animation is always triggered on the last item of the list

### Expected behavior

The animation should be triggered on the new item, not on the last item of the list

**Comment by MrBrax** [comment]
i think this would be related as a fix? #4103

**Comment by bub-bl** [comment]
> i think this would be related as a fix? [#4103](https://github.com/Facepunch/sbox-issues/issues/4103)

Yes, we need to up this kind of issues a little bit more, i get a lot of issue with all the ui stuff

**Comment by solwllms** [comment]
We support the `@key` attribute now, so if you add `@key=item` to the span in your example snippet it should behave like you want.
